library/x509_crt.c

```diff
@@ -2179,13 +2179,6 @@ static int x509_crt_find_parent(
         *parent_is_trusted = 0;
     }
 
-    /* extra precaution against mistakes in the caller */
-    if( parent == NULL )
```

Comment:
@mpg @k-stachowiak Was this check meant to be `if( *parent == NULL )` in the first place, clearing the destination variables in case no parent was found?

Reply:
I think the intention was indeed to check for `*parent == NULL`. @k-stachowiak can you please amend your PR to change this check to use `*parent` instead of removing it? Thanks!